### DISCUSSION TITLE
Hitting local `/api` from `getServerSideProps`

### DISCUSSION THREAD

**Discussion by jackHedaya** [vercel/next.js#20962] [Help]
Hey!

I was wondering what the recommended method of hitting `/api` is from `getServerSideProps`. Of course, just fetching it is an option but it's definitely better to avoid a round trip. Can I import the function from `/api` and call it by passing in `ctx.req` and `ctx.res`? Is it safe to cast them considering `req` is of type `IncomingMessage` and the `/api` accepts `NextApiRequest`?

**Comment by michrome** [comment]
My recommendation is to extract the code/logic that's running in `/api` into a separate js file that is called by both your API and `getServerSideProps`.

It _feels_ inelegant to be calling the API end points during the deploy because it may not be clear which version is being called. I think if you make a call to an API end point during your build, you won't be calling the version you're deploying … you'll be calling the version from the last successful build.
